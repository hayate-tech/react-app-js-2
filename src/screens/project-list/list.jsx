export const List = ({ list, managers }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Project Name</th>
          <th>Manager</th>
        </tr>
      </thead>
      <tbody>
        {list.map((project) => (
          <tr key={project.id}>
            <td>{project.name}</td>
            <td>
              {managers.find((manager) => manager.id === project.manager_id)
                ?.name || "unknown"}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};
