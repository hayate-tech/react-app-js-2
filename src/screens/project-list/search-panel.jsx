// import { useState, useEffect } from "react"
import _ from "lodash";
import React from "react";
import { Input, Select } from "antd";

const { Option } = Select;

const SearchSelect = React.memo(({ setManagerId, managerId, managers }) => {
  return (
    <>
      <Select
        defaultValue={managerId}
        onChange={(value) => {
          setManagerId(value);
        }}
      >
        <Option value="">Manager</Option>
        {managers.map((manager) => (
          <Option value={manager.id} key={manager.id}>
            {manager.name}
          </Option>
        ))}
      </Select>
    </>
  );
});

export const SearchPanel = ({
  projectName,
  setProjectName,
  managerId,
  setManagerId,
  managers,
}) => (
  <form>
    <Input
      type="text"
      value={projectName}
      onChange={(evt) => {
        setProjectName(evt.target.value);
      }}
    />
    <SearchSelect
      setManagerId={setManagerId}
      managerId={managerId}
      managers={managers}
    />
  </form>
);
