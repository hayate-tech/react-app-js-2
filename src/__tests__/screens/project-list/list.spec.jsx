import React from "react";
import { Table } from "antd";
import { List } from "../../../screens/project-list/list";
import { shallow, mount } from "enzyme";

describe("../../screens/project-list/list", () => {
  const makeProps = (override) => ({
    list: [
      {
        id: 1,
        name: "Project1",
        manager_id: 3,
        tream_id: "Team One",
        created_id: 982396349,
      },
    ],
    managers: [{ id: 3, name: "Vincent" }],
    ...override,
  });

  it("should render table by default", () => {
    const props = makeProps();
    const wrapper = shallow(<List {...props} />);

    expect(wrapper.find(Table)).toHaveLength(1);

    expect(wrapper.find(Table).props().columns).toEqual([
      {
        title: "Project Name",
        dataIndex: "project_name",
        key: "project_name",
      },
      { title: "Manager", dataIndex: "manager", key: "manager" },
    ]);
  });

  it("should render table", () => {
    const props = makeProps();
    const wrapper = mount(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render table when manager's id is not equal manager's in the project", () => {
    const props = makeProps({
      list: [
        {
          id: 1,
          name: "Project1",
          manager_id: 1,
          tream_id: "Team One",
          created_id: 982396349,
        },
      ],
    });
    const wrapper = shallow(<List {...props} />);

    expect(wrapper.find(Table)).toHaveLength(1);

    expect(wrapper.find(Table).props().dataSource).toEqual([
      { key: 1, project_name: "Project1", manager: "unknown" },
    ]);
  });
});
