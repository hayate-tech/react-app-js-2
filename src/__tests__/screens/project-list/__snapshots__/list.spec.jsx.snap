// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../../screens/project-list/list should render table 1`] = `ReactWrapper {}`;
